Explicitly setting the color profile

Curious how y'all test the lipgloss output. I'm currently running into issues where the terminal colors are sometimes rendered in tests and sometimes they are not, and it seems to be based on which directory I'm running `go test` from. Any ideas on how to handle this?

<img width="950" alt="Screen Shot 2021-05-26 at 10 16 31 AM" src="https://user-images.githubusercontent.com/24326455/119703506-74e4c700-be0b-11eb-9f15-dd726e347ba9.png">

Indeed! Here's a minimum repro:


### Code
repro.go
```go
package main

import (
	"fmt"
	"github.com/charmbracelet/lipgloss"
)

var myStyle = lipgloss.NewStyle().Italic(true).Foreground(lipgloss.Color("205"))

func colorString(str string) string {
	return myStyle.Render(str)
}

func main() {
	fmt.Println(colorString("hello world"))
}
```

repro_test.go
```go
package main

import (
	"strings"
	"testing"
)

func Test_colorString(t *testing.T) {
	tests := []struct {
		input string
		want string
	}{
		{ "hello world", "ESC[3;38;5;205mhello worldESC[0m"},
	}
		for _, tt := range tests {
		t.Run(tt.input, func(t *testing.T) {
			got := colorString(tt.input)
			if got = strings.ReplaceAll(got, "\x1b", "ESC"); got != tt.want {
				t.Errorf("colorString() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

### Instructions
1. Run `go test` - observe success
```
❯ go test  
PASS
ok      lipgloss-repro  0.164s
```
2. Run `go test .` - observe failure
```
❯ go test .
--- FAIL: Test_colorString (0.00s)
    --- FAIL: Test_colorString/hello_world (0.00s)
        repro_test.go:20: colorString() = ESC[3;mhello worldESC[0m, want ESC[3;38;5;205mhello worldESC[0m
FAIL
FAIL    lipgloss-repro  0.334s
FAIL
```

--- a/README.md
+++ b/README.md
@@ -51,13 +51,15 @@ lipgloss.Color("201") // hot pink
 lipgloss.Color("202") // orange
 ```
 
-### True Color (24-bit)
+### True Color (16,777,216 colors; 24-bit)
 
 ```go
 lipgloss.Color("#0000FF") // good ol' 100% blue
 lipgloss.Color("#04B575") // a green
 lipgloss.Color("#3C3C3C") // a dark gray
 ```
+
+...as well as a 1-bit Ascii profile, which is black and white only.
 
 The terminal's color profile will be automatically detected, and colors outside
 the gamut of the current palette will be automatically coerced to their closest

--- a/color.go
+++ b/color.go
@@ -8,8 +8,10 @@ import (
 )
 
 var (
-	colorProfile    termenv.Profile
-	getColorProfile sync.Once
+	colorProfile         termenv.Profile
+	getColorProfile      sync.Once
+	explicitColorProfile bool
+	colorProfileMtx      sync.Mutex
 
 	// Because it's a potentially long operation (relatively speaking), we
 	// check the background color on initialization rather than at the last
@@ -20,10 +22,35 @@ var (
 // ColorProfile returns the detected termenv color profile. It will perform the
 // actual check only once.
 func ColorProfile() termenv.Profile {
-	getColorProfile.Do(func() {
-		colorProfile = termenv.ColorProfile()
-	})
+	if !explicitColorProfile {
+		getColorProfile.Do(func() {
+			colorProfile = termenv.ColorProfile()
+		})
+	}
 	return colorProfile
+}
+
+// SetColorProfile sets the color profile on a package-wide context. This
+// function exists mostly for testing purposes so that you can assure you're
+// testing against a specific profile.
+//
+// Outside of testing you likely won't want to use this function as
+// ColorProfile() will detect and cache the terminal's color capabilities
+// and choose the best available profile.
+//
+// Available color profiles are:
+//
+// termenv.Ascii (no color, 1-bit)
+// termenv.ANSI (16 colors, 4-bit)
+// termenv.ANSI256 (256 colors, 8-bit)
+// termenv.TrueColor (16,777,216 colors, 24-bit)
+//
+// This function is thread-safe.
+func SetColorProfile(p termenv.Profile) {
+	colorProfileMtx.Lock()
+	defer colorProfileMtx.Unlock()
+	colorProfile = p
+	explicitColorProfile = true
 }
 
 // HadDarkBackground returns whether or not the terminal has a dark background.

--- a/color_test.go
+++ b/color_test.go
@@ -1,0 +1,53 @@
+package lipgloss
+
+import (
+	"testing"
+
+	"github.com/muesli/termenv"
+)
+
+func TestSetColorProfile(t *testing.T) {
+	t.Parallel()
+
+	tt := []struct {
+		profile  termenv.Profile
+		input    string
+		style    Style
+		expected string
+	}{
+		{
+			termenv.Ascii,
+			"hello",
+			NewStyle().Foreground(Color("#5A56E0")),
+			"hello",
+		},
+		{
+			termenv.ANSI,
+			"hello",
+			NewStyle().Foreground(Color("#5A56E0")),
+			"\x1b[34mhello\x1b[0m",
+		},
+		{
+			termenv.ANSI256,
+			"hello",
+			NewStyle().Foreground(Color("#5A56E0")),
+			"\x1b[38;5;62mhello\x1b[0m",
+		},
+		{
+			termenv.TrueColor,
+			"hello",
+			NewStyle().Foreground(Color("#5A56E0")),
+			"\x1b[38;2;89;86;224mhello\x1b[0m",
+		},
+	}
+
+	for i, tc := range tt {
+		SetColorProfile(tc.profile)
+		res := tc.style.Render(tc.input)
+		if res != tc.expected {
+			t.Errorf("Test %d, expected:\n\n`%s`\n`%s`\n\nActual output:\n\n`%s`\n`%s`\n\n",
+				i, tc.expected, formatEscapes(tc.expected),
+				res, formatEscapes(res))
+		}
+	}
+}

--- a/runes_test.go
+++ b/runes_test.go
@@ -58,13 +58,5 @@ func TestStyleRunes(t *testing.T) {
 }
 
 func formatEscapes(str string) string {
-	var b strings.Builder
-	for _, r := range str {
-		if r == '\x1b' {
-			b.WriteString("\\x1b")
-			continue
-		}
-		b.WriteRune(r)
-	}
-	return b.String()
+	return strings.ReplaceAll(str, "\x1b", "\\x1b")
 }

